### ISSUE TITLE
Cannot set maximum_throughput_units when auto_inflate_enabled is false

### ISSUE THREAD

**Issue by KenticoMartinS** [terraform-providers/terraform-provider-azurerm#2083]
Hi,
we are running Terraform with Azure provider module:
```
Terraform v0.11.8
+ provider.azurerm v1.16.0
```

We have a general module containing Event Hub Namespace resource
```
resource "azurerm_eventhub_namespace" "default" {
  name                     = "${var.event_hub_namespace_name}"
  location                 = "${var.location}"
  resource_group_name      = "${var.resource_group_name}"
  auto_inflate_enabled     = "${var.autoscaling["enabled"]}"
  maximum_throughput_units = "${var.autoscaling["max_units"]}"

  sku      = "${var.sku}"
  capacity = "${var.capacity}"
}
```
From the module, we are trying to set **auto_inflate_enabled** and **maximum_throughput_units** parameters. The **maximum_throughput_units** parameter is dependent on **auto_inflate_enabled** parameters. It means, we can set **maximum_throughput_units** only when **auto_inflate_enabled** is enabled.

Is it possible to use one **azurerm_eventhub_namespace** resource for the following two scenarios?
1. **auto_inflate_enabled** is disabled and we don't care about **maximum_throughput_units**
2. **auto_inflate_enabled** is enabled and **maximum_throughput_units** is set to e.g. 20

We tried the following solutions:
1. By default set **auto_inflate_enabled** to false and **maximum_throughput_units** to 0. Following error occured: `Error: module.event_hub.azurerm_eventhub_namespace.default: expected maximum_throughput_units to be in the range (1 - 20), got 0`
2. By default set **auto_inflate_enabled** to false and **maximum_throughput_units** to 1. Following error occured: `eventhub.NamespacesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="Cannot set MaximumThroughputUnits property if AutoInflate is not enabled.`

**Comment by KenticoMartinS** [comment]
Hi @katbyte,
thank you very much for looking into the issue. We are looking forward to the fix because currently, it is blocking us from creating a general module containing the Event Hub resource.

**Comment by tombuildsstuff** [comment]
hi @KenticoMartinS 


Just to let you know that this has been released as a part of v1.18 of the AzureRM Provider ([the full changelog is available here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.18.0/CHANGELOG.md)). You can upgrade to this by specifying the version in the provider block (as shown below) and then running `terraform init -upgrade`

```
provider "azurerm" {
  version = "=1.18.0"
}
```

Thanks!
